docker-compose-separate-containers.yml

```diff
@@ -1,23 +1,25 @@
-version: '2'
+version: '3'
 services:
   nginx:
     image: nginx
     container_name: nginx
     ports:
       - "80:80"
     volumes:
-      - /etc/nginx/conf.d
+      - nginx_conf:/etc/nginx/conf.d
 
   dockergen:
     image: jwilder/docker-gen
     command: -notify-sighup nginx -watch /etc/docker-gen/templates/nginx.tmpl /etc/nginx/conf.d/default.conf
-    volumes_from:
-      - nginx
     volumes:
       - /var/run/docker.sock:/tmp/docker.sock:ro
       - ./nginx.tmpl:/etc/docker-gen/templates/nginx.tmpl
+      - nginx_conf:/etc/nginx/conf.d
```

Comment:
Are you sure adding `nocopy` doesn't fix it? Because it works for me. You'll need to make sure to remove the volumes before each run to be able to test it properly. `depends_on` has no effect when using `docker stack deploy`, so it's not a solution.